core/src/test/java/org/elasticsearch/cluster/MinimumMasterNodesIT.java

```diff
@@ -278,20 +276,23 @@ public void dynamicUpdateMinimumMasterNodes() throws Exception {
     }
 
     private void assertNoMasterBlockOnAllNodes() throws InterruptedException {
-        assertThat(awaitBusy(new Predicate<Object>() {
-            @Override
-            public boolean apply(Object obj) {
-                boolean success = true;
-                for (Client client : internalCluster()) {
-                    ClusterState state = client.admin().cluster().prepareState().setLocal(true).execute().actionGet().getState();
-                    success &= state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID);
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Checking for NO_MASTER_BLOCK on client: {} NO_MASTER_BLOCK: [{}]", client, state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID));
-                    }
-                }
-                return success;
-            }
-        }, 20, TimeUnit.SECONDS), equalTo(true));
+        assertThat(awaitBusy(
+                        () -> {
+                            boolean success = true;
+                            for (Client client : internalCluster()) {
+                                ClusterState state = client.admin().cluster().prepareState().setLocal(true).execute().actionGet().getState();
+                                success &= state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID);
+                                if (logger.isDebugEnabled()) {
+                                    logger.debug("Checking for NO_MASTER_BLOCK on client: {} NO_MASTER_BLOCK: [{}]", client, state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID));
+                                }
+                            }
+                            return success;
+                        },
+                        20,
+                        TimeUnit.SECONDS
```

Comment:
@jpountz Do you think that the change in 6a5e23f6feb887775699eff90dcae95c99f1c435 is better? If that logging statement weren't there, and the loop were short-circuiting (which might only be not short-circuiting because of the logging statement) I think that it could be improved further. Do you see any other opportunities to improve?